### PR TITLE
More rover ingestion

### PR DESCRIPTION
**Pull request by leogrim**
@andrewconner @stkem @yingjieMiao

### CODE REVIEW

**Comment by stkem** [comment]
:+1:

File: kifi-backend/modules/common/conf/evolutions/shoebox/300.sql [hunk 1]
```diff
@@ -16,10 +16,12 @@ CREATE TABLE if not exists article_info (
     latest_version_minor int NULL,
     oldest_version_major int NULL,
     oldest_version_minor int NULL,
-    last_queued_at datetime NULL,
     last_fetched_at datetime NULL,
     next_fetch_at datetime NULL,
-    fetch_interval float NOT NULL,
+    fetch_interval bigint(20) NULL,
```
Comment (yingjieMiao) [inline kifi-backend/modules/common/conf/evolutions/shoebox/300.sql:21]: `bigint` seems very big for this field.

Comment (yingjieMiao) [inline kifi-backend/modules/common/conf/evolutions/shoebox/300.sql:21]: oh, did you use seconds as unit? too fine grained? hours should be enough?

File: kifi-backend/modules/rover/app/com/keepit/rover/manager/FetchSchedulingPolicy.scala [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  def nextFetchAfterFailure(failureCount: Int): DateTime = {
+    val backoff = (backoffIncrement * (1 << failureCount)) min maxBackoff
+    val secondsToNextFetch = (backoff + randomDelay).toSeconds.toInt
+    currentDateTime plusSeconds secondsToNextFetch
+  }
+
+  def nextFetchAfterSuccess(interval: Duration): DateTime = {
+    val secondsToNextFetch = (interval + randomDelay).toSeconds.toInt
+    currentDateTime plusMinutes secondsToNextFetch
+  }
```
Comment (yingjieMiao) [inline kifi-backend/modules/rover/app/com/keepit/rover/manager/FetchSchedulingPolicy.scala:33]: did you mean `plusSeconds` here?

Comment (leogrim) [inline kifi-backend/modules/rover/app/com/keepit/rover/manager/FetchSchedulingPolicy.scala:33]: Yep thanks!

File: kifi-backend/modules/rover/app/com/keepit/rover/manager/FetchSchedulingPolicy.scala [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+  def increaseInterval(currentInterval: Duration): Duration = (currentInterval - intervalDecrement) max minInterval
+  def decreaseInterval(currentInterval: Duration): Duration = (currentInterval + intervalIncrement) min maxInterval
+}
```
Comment (yingjieMiao) [inline kifi-backend/modules/rover/app/com/keepit/rover/manager/FetchSchedulingPolicy.scala:37]: a bit confused with the semantic here:
`increaseInterval` corresponds to `- intervalDecrement`
`decreaseInterval` corresponds to `+interavlIncrement`

either typo or confusing naming, or I missed some context

File: kifi-backend/modules/common/conf/evolutions/shoebox/300.sql (hunk 1 above)
Comment (leogrim) [inline kifi-backend/modules/common/conf/evolutions/shoebox/300.sql:21]: It's a `Long` in Scala, the serialization is using milliseconds.

Comment (yingjieMiao) [inline kifi-backend/modules/common/conf/evolutions/shoebox/300.sql:21]: ok

File: kifi-backend/modules/rover/app/com/keepit/rover/manager/FetchSchedulingPolicy.scala (hunk 3 above)
Comment (yingjieMiao) [inline kifi-backend/modules/rover/app/com/keepit/rover/manager/FetchSchedulingPolicy.scala:37]: Another note, it perhaps safer to explicitly make sure the resulting interval is within `[minValue, maxValue]`, 
e.g something like `(x max minValue) min maxValue`.

The old scraper service had a related bug.

**Comment by yingjieMiao** [comment]
a few minor comments, lgtm overall. Some other reviews might be helpful.

File: kifi-backend/modules/rover/app/com/keepit/rover/manager/FetchSchedulingPolicy.scala (hunk 3 above)
Comment (leogrim) [inline kifi-backend/modules/rover/app/com/keepit/rover/manager/FetchSchedulingPolicy.scala:37]: Good catch + good suggestion, all fixed :)

**Comment by leogrim** [comment]
I'll merge this before adding the queuing in.

**Pull request merged**